### PR TITLE
Plans 2023: Remove usePlanPrices abstraction and memoize getPlanPrices one

### PR DESCRIPTION
**Pull request by chriskmnds**
## Proposed Changes

- Removes the `usePlanPrices` hook as it's merely calling the selector underneath
- Memoizes the `getPlanPrices` selector to ensure referential integrity for same-arguments calls

Technically, both of these abstractions are unnecessary, especially since the introduction of `usePricingMetaForGridPlans`. However, the alternative (not having either) does lead to slightly more boilerplate to grab the prices (and will also require a similar but local selector in `usePricingMetaForGridPlans`). 

The cost for keeping the selector is memoization to ensure same reference on consecutive/same-argument calls. This should take care of some of the Redux warnings that pop up in the console. I am not sure of any other considerable gains right now from memoizing this selector, outside of referential integrity - which will allow us to optimize further transformations and components (I think memoizing this selector may actually be unnecessary for serving https://github.com/Automattic/wp-calypso/pull/82463).

## Testing Instructions

* Go to `/plans` and `/start/plans`
    * ensure pricing shows as expected
* Go to `/start/onboarding-pm` as the only consumer of `usePlanPrices`
    * choose free domain and free plan. ensure the popup shows the correct pricing info

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/plans/selectors/get-plan-prices.ts [hunk 1]
```diff
@@ -16,36 +16,44 @@ export interface PlanPrices {
...
+					  } )
+					: null,
+		};
+	},
+	( state: AppState, { siteId } ) => [
+		getPlansBySiteId( state, siteId ?? undefined ), // consumed by getPlanDiscountedRawPrice
+		state.plans?.items, // consumed by getPlanRawPrice, getDiscountedRawPrice
+	],
+	( state: AppState, { planSlug, siteId, returnMonthly, returnSmallestUnit } ) =>
+		planSlug + siteId + returnMonthly + returnSmallestUnit
```
Comment (chriskmnds) [inline client/state/plans/selectors/get-plan-prices.ts:58]: This is the cache key. First time I use this, and not too fond of it living hidden here. It probably needs narrowing down too.

Not sure how else to get the dependencies array above to include non-state elements.

**Comment by chriskmnds** [comment]
> Technically, both of these abstractions are unnecessary, especially since the introduction of `usePricingMetaForGridPlans`. However, the alternative (not having either) does lead to slightly more boilerplate to grab the prices (and will also require a similar but local selector in `usePricingMetaForGridPlans`). 

p.s. In this initial/first commit https://github.com/Automattic/wp-calypso/pull/83357/commits/bd76438b91fee9d47aa782582cdc5018677b063e we completely remove the use of `getPlanPrices`. There's more going on depending on how one sees it. These are also subject to being refactored in a later milestone on the NPM Migration PT. I'm personally okay with keeping the abstraction until later.

**Review by ddc22 — APPROVED** [review]
LGTM 🚢

**Comment by chriskmnds** [comment]
Thanks for taking the time!

**Pull request merged**